agent/retry_join.go

```diff
@@ -115,15 +124,18 @@ func (r *retryJoiner) retryJoin() error {
 		return err
 	}
 
-	r.logger.Printf("[INFO] agent: Retry join %s is supported for: %s", r.cluster, strings.Join(disco.Names(), " "))
-	r.logger.Printf("[INFO] agent: Joining %s cluster...", r.cluster)
+	r.logger.Info("Retry join is supported for the following discovery methods",
+		"cluster", r.cluster,
```

Comment:
Isn't this `cluster` field already present due to the `With("cluster",...)` used when the derived logger is created?